src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,108 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
+                                    $"does not exist as a column in {entityName}.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+
+                    foreach (string targetField in relationship.TargetFields)
+                    {
+                        if (!sqlMetadataProvider.TryGetBackingColumn(relationship.TargetEntity, targetField, out _))
+                        {
+                            // Validation to ensure that entities that have target fields defined, define target fields
+                            // that are valid columns in the target entity.
+                            HandleOrRecordException(new DataApiBuilderException(
+                                message: $"Entity: {entityName} has a relationship: {relationshipName} with target field: {targetField} that " +
+                                    $"does not exist as a column in target entity: {relationship.TargetEntity}.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+                }
+
+                if (relationship.LinkingSourceFields is not null && relationship.LinkingTargetFields is null)
+                {
+                    // Validation to ensure that if linking source fields exist that linking target fields exist as well.
+                    HandleOrRecordException(new DataApiBuilderException(
+                        message: $"Entity: {entityName} has a relationship: {relationshipName} with linking source fields that are not null, " +
+                            $"but linking target fields that are null.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                }
+
+                if (relationship.LinkingTargetFields is not null && relationship.LinkingSourceFields is null)
+                {
+                    // Validation to ensure that if linking target fields exist that linking source fields exist as well.
+                    HandleOrRecordException(new DataApiBuilderException(
+                        message: $"Entity: {entityName} has a relationship: {relationshipName} with linking target fields that are not null, " +
+                            $"but linking source fields that are null.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                }
+
+                if (relationship.LinkingSourceFields is not null && relationship.LinkingTargetFields is not null)
```

Comment:
asymmetrical existence to be moved to rc2 because this info may be resolved from FKDefinition resolved from db

Reply:
fixed by removal. deferred to rc2